py_trees/decorators.py

```diff
@@ -83,6 +83,7 @@ def __init__(self, child, name=common.Name.AUTO_GENERATED):
         # Initialise
         super(Decorator, self).__init__(name=name)
         self.children.append(child)
+        child.parent = self
```

Comment:
Could you push this up a line? I know python appends by reference, but the logical flow reads more comfortably that way (and it helps prevent you getting into habits that will catch you out in other languages).